gulpfile.js

```diff
@@ -39,80 +47,130 @@ gulp.task('lint', function() {
       `${APP_ROOT}**/*.{js,jsx}`,
       `${TESTS_ROOT}**/*.js`,
       './*.js',
-      `!${APP_ROOT}components/**`,
     ])
     .pipe(eslint())
     .pipe(eslint.format())
     .pipe(eslint.failAfterError());
 });
 
 /**
- * Copies necessary files for the Babel amd loader to the app.
+ * Copy all non-js directory app source/assets/components.
  */
-gulp.task('loader-polyfill', function() {
-  return gulp.src([
-      './node_modules/alameda/alameda.js',
-      './node_modules/whatwg-fetch/fetch.js',
-      './node_modules/url-search-params/build/url-search-params.max.js',
-      `${APP_ROOT}js/bootstrap.js`,
-    ])
-    .pipe(concat('initapp.js'))
-    .pipe(gulp.dest(`${DIST_APP_ROOT}js`));
+gulp.task('copy-app', function() {
+  return merge(
+    gulp.src([
+      `${APP_ROOT}**`,
+      // Don't copy documentation files.
+      `!${APP_ROOT}**/*.md`,
+      // Don't copy JS, it will be compiled and copied on the compile step.
+      `!${APP_ROOT}js/**`,
+    ]),
+
+    // Module loader.
+    gulp.src('./node_modules/alameda/alameda.js')
+      .pipe(rename('js/alameda.js')),
+
+    // Components.
+    gulp.src('./node_modules/react/dist/react.min.js')
+      .pipe(rename('js/components/react.js')),
+    gulp.src('./node_modules/react-dom/dist/react-dom.min.js')
+      .pipe(rename('js/components/react-dom.js')),
+    gulp.src('./node_modules/fxos-mvc/dist/mvc.js')
+      .pipe(rename('js/components/mvc.js')),
```

Comment:
We still want to keep the non minified files for React in debug builds because they comes with additional checking and clever warning useful for debugging.